tests/system/feature_store/test_spark_engine.py

```diff
@@ -1444,20 +1444,22 @@ def test_error_is_properly_propagated(self):
 
     # ML-5726
     @pytest.mark.skipif(
-        not {"HDFS_HOST", "HDFS_PORT", "HDFS_HTTP_PORT"}.issubset(os.environ.keys()),
-        reason="HDFS host and ports are not defined",
+        not {"HDFS_HOST", "HDFS_PORT", "HDFS_HTTP_PORT", "HADOOP_USER_NAME"}.issubset(
+            os.environ.keys()
+        ),
+        reason="HDFS host, ports and user name are not defined",
     )
     def test_ingest_and_get_offline_features_with_hdfs(self):
         key = "patient_id"
 
-        register_temporary_client_datastore_profile(
-            DatastoreProfileHdfs(
-                name="my-hdfs",
-                host=os.getenv("HDFS_HOST"),
-                port=int(os.getenv("HDFS_PORT")),
-                http_port=int(os.getenv("HDFS_HTTP_PORT")),
-            )
+        datastore_profile = DatastoreProfileHdfs(
+            name="my-hdfs",
+            host=os.getenv("HDFS_HOST"),
+            port=int(os.getenv("HDFS_PORT")),
+            http_port=int(os.getenv("HDFS_HTTP_PORT")),
         )
+        register_temporary_client_datastore_profile(datastore_profile)
+        self.project.register_datastore_profile(datastore_profile)
```

Comment:
@alxtkr77, it's apparently necessary to call `register_temporary_client_datastore_profile` in addition to `register_datastore_profile`. Is that by design?

Reply:
When running locally (not from pod) the mlrun has no access to secrets, so you have to call register_temporary_client_datastore_profile()

Reply:
If this is really intended usage I think we should probably document it. From what I can tell, you always have to call both before calling a remote `get_offline_features` (e.g. with spark engine). Or we could have `register_datastore_profile` call `register_temporary_client_datastore_profile` internally.